### PR TITLE
⚠️  add v1beta1 support

### PR DESCRIPTION
**Pull request by roytman**
Background of the PR.
We have published the v1.0.0 but did not update the CRDs apiVersions. According to Kubernetes backward compatibility starts from `beta` versions. Therefore, we have to promote our CRD versions to `v1beta1` 

Signed-off-by: Alexey Roytman <roytman@il.ibm.com>

Co-authored-by: Alexey Roytman <roytman@il.ibm.com>
Co-authored-by: Raviv Schaffer <raviv.schaffer@ibm.com>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -88,7 +89,7 @@ test: pre-test
 run-integration-tests: export DOCKER_HOSTNAME?=localhost:5000
 run-integration-tests: export DOCKER_NAMESPACE?=fybrik-system
 run-integration-tests: export VALUES_FILE=charts/fybrik/integration-tests.values.yaml
-run-integration-tests: export HELM_SETTINGS=--set "manager.solver.enabled=true"
+run-integration-tests: export HELM_SETTINGS=--set "manager.solver.enabled=false "
```
Comment (shlomitk1) [inline Makefile:92]: Please leave "true". We run integration tests with solver enabled. 
If solver is not integrated (the corresponding image is empty), the enabled flag won't matter.

File: pkg/multicluster/razee/razee_cluster_manager.go [hunk 2]
```diff
@@ -21,15 +21,15 @@ import (
...
+	app "fybrik.io/fybrik/manager/apis/app/v1beta1"
 	"fybrik.io/fybrik/pkg/logging"
 	"fybrik.io/fybrik/pkg/multicluster"
 )
 
 const (
 	clusterMetadataConfigMapSL = "/api/v1/namespaces/fybrik-system/configmaps/cluster-metadata"
 	endPointURL                = "https://config.satellite.cloud.ibm.com/graphql"
-	bluePrintSelfLink          = "/apis/app.fybrik.io/v1alpha1/namespaces/%s/blueprints/%s"
+	bluePrintSelfLink          = "/apis/app.fybrik.io/app/namespaces/%s/blueprints/%s"
```
Comment (shlomitk1) [inline pkg/multicluster/razee/razee_cluster_manager.go:32]: Should be v1beta1

File: manager/apis/app/v1beta1/fybrikapplication_types.go [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []FybrikApplication `json:"items"`
+}
+
+func init() {
+	SchemeBuilder.Register(&FybrikApplication{}, &FybrikApplicationList{})
+}
+
+const (
```
Comment (shlomitk1) [inline manager/apis/app/v1beta1/fybrikapplication_types.go:236]: These definitions should be removed from v1alpha1 package.

File: manager/apis/app/v1beta1/fybrikapplication_webhook.go [hunk 4]
```diff
@@ -0,0 +1,71 @@
+// Copyright 2020 IBM Corp.
+// SPDX-License-Identifier: Apache-2.0
+
+package v1beta1
+
```
Comment (shlomitk1) [inline manager/apis/app/v1beta1/fybrikapplication_webhook.go:5]: The parallel file in v1alpha1 is not needed (main uses only v1beta1 webhook for fybrikapplications)

File: manager/apis/app/v1beta1/fybrikapplication_webhook_test.go [hunk 5]
```diff
@@ -0,0 +1,99 @@
+// Copyright 2020 IBM Corp.
+// SPDX-License-Identifier: Apache-2.0
+
+package v1beta1
+
```
Comment (shlomitk1) [inline manager/apis/app/v1beta1/fybrikapplication_webhook_test.go:5]: Same comment - the parallel file for v1alpha1 is not needed.

File: manager/apis/app/v1beta1/helm_values.go [hunk 6]
```diff
@@ -0,0 +1,16 @@
+// Copyright 2020 IBM Corp.
+// SPDX-License-Identifier: Apache-2.0
+
+package v1beta1
+
+// HelmValues are the values passed to modules during orchestration of the data plane
+type HelmValues struct {
```
Comment (shlomitk1) [inline manager/apis/app/v1beta1/helm_values.go:7]: The parallel file in v1alpha1 is not used.

File: manager/apis/app/v1beta1/interface_types.go [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+// SPDX-License-Identifier: Apache-2.0
+
+package v1beta1
+
+import (
+	"fybrik.io/fybrik/pkg/model/taxonomy"
+)
+
+// connection types and formats used in tests and mockup connectors
+const (
```
Comment (shlomitk1) [inline manager/apis/app/v1beta1/interface_types.go:11]: These definitions are not used in v1alpha1 and should be removed.

File: Makefile [hunk 8]
```diff
@@ -16,6 +16,7 @@ license: $(TOOLBIN)/license_finder
 .PHONY: generate
 generate: $(TOOLBIN)/controller-gen $(TOOLBIN)/json-schema-generator
 	$(TOOLBIN)/json-schema-generator -r ./manager/apis/app/v1alpha1/ -o charts/fybrik/files/taxonomy/
+	$(TOOLBIN)/json-schema-generator -r ./manager/apis/app/v1beta1/ -o charts/fybrik/files/taxonomy/
```
Comment (shlomitk1) [inline Makefile:19]: Why is the old one used for generating taxonomy files? There will be only one version of the file in taxonomy.

File: Makefile (hunk 1 above)
Comment (roytman) [reply Makefile:92]: I cannot run solver on my laptop, so I set it to false, and mistakenly committed it. Thanks

File: manager/apis/app/v1beta1/fybrikapplication_types.go (hunk 3 above)
Comment (roytman) [reply manager/apis/app/v1beta1/fybrikapplication_types.go:236]: The better to move all this constants to controllers, they are not part of APIs.

File: manager/apis/app/v1beta1/fybrikapplication_webhook.go (hunk 4 above)
Comment (roytman) [reply manager/apis/app/v1beta1/fybrikapplication_webhook.go:5]: No we need webhooks for both versions and main uses all of them.

File: manager/apis/app/v1beta1/fybrikapplication_webhook_test.go (hunk 5 above)
Comment (roytman) [reply manager/apis/app/v1beta1/fybrikapplication_webhook_test.go:5]: The same answer.

File: manager/apis/app/v1beta1/helm_values.go (hunk 6 above)
Comment (roytman) [reply manager/apis/app/v1beta1/helm_values.go:7]: OK.

File: manager/apis/app/v1beta1/interface_types.go (hunk 7 above)
Comment (roytman) [reply manager/apis/app/v1beta1/interface_types.go:11]: OK

File: Makefile (hunk 8 above)
Comment (roytman) [reply Makefile:19]: In this case, yes, the old version can be removed. In the future, if APIVersion's schemas can be different, we will have hard time with taxonomy.  
Removed.

File: manager/apis/app/v1beta1/helm_values.go (hunk 6 above)
Comment (roytman) [reply manager/apis/app/v1beta1/helm_values.go:7]: @shlomitk1 , maybe will be better to store the helm values in some other place. They are not part of K8s API.

File: manager/apis/app/v1beta1/interface_types.go (hunk 7 above)
Comment (roytman) [reply manager/apis/app/v1beta1/interface_types.go]: The same comment about location of the constants.

File: manager/apis/app/v1beta1/helm_values.go (hunk 6 above)
Comment (shlomitk1) [reply manager/apis/app/v1beta1/helm_values.go:7]: They were in other place but we wanted to make them visible.

File: manager/apis/app/v1beta1/fybrikapplication_webhook_test.go (hunk 5 above)
Comment (shlomitk1) [reply manager/apis/app/v1beta1/fybrikapplication_webhook_test.go:5]: Ok, I missed that.

File: manager/apis/app/v1beta1/fybrikapplication_webhook.go (hunk 4 above)
Comment (shlomitk1) [reply manager/apis/app/v1beta1/fybrikapplication_webhook.go:5]: Right, I did not see the usage.

File: manager/apis/app/v1beta1/fybrikapplication_types.go (hunk 3 above)
Comment (shlomitk1) [reply manager/apis/app/v1beta1/fybrikapplication_types.go:236]: They can't be in the controllers package because of the import loop, maybe somewhere under pkg/

File: manager/controllers/app/fybrik_notebook_write_flow_test.go [hunk 9]
```diff
@@ -30,7 +30,7 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 
 	"fybrik.io/fybrik/connectors/katalog/pkg/apis/katalog/v1alpha1"
```
Comment (revit13) [inline manager/controllers/app/fybrik_notebook_write_flow_test.go:32]: katalog/v1alpha1 -> katalog/v1beta1?

Comment (roytman) [reply manager/controllers/app/fybrik_notebook_write_flow_test.go:32]: I did not promote Katalog's API, it is our proprietary interface, so I think it's better to leave it in the alpha stage.

File: pkg/multicluster/razee/razee_cluster_manager.go (hunk 2 above)
Comment (roytman) [reply pkg/multicluster/razee/razee_cluster_manager.go:32]: done, thanks.

File: manager/apis/app/v1beta1/helm_values.go (hunk 6 above)
Comment (roytman) [reply manager/apis/app/v1beta1/helm_values.go:7]: I suggest moving them to `pkg/helm`

**Comment by simanadler** [comment]
@roytman (1) Do we have a design doc describing how versioning is done, our overall approach to handling versions, etc?  ex: will the version numbers of different CRDs be aligned? (2) why are we moving to beta instead of 1.0?

File: charts/fybrik/files/taxonomy/external.json [hunk 10]
```diff
@@ -0,0 +1,339 @@
+{
```
Comment (shlomitk1) [inline charts/fybrik/files/taxonomy/external.json:1]: Why is this file committed?

File: charts/fybrik/files/webhook-configs.yaml [hunk 11]
```diff
@@ -14,14 +14,14 @@ webhooks:
       service:
         name: webhook-service
         namespace: '{{ .Release.Namespace }}'
-        path: /validate-app-fybrik-io-v1alpha1-fybrikapplication
+        path: /validate-app-fybrik-io-v1-fybrikapplication
     failurePolicy: Fail
-    name: vfybrikapplication.kb.io
+    name: vfybrikapplication-beta.kb.io
```
Comment (shlomitk1) [inline charts/fybrik/files/webhook-configs.yaml:19]: v1?

File: charts/fybrik/files/webhook-configs.yaml [hunk 12]
```diff
@@ -35,14 +35,14 @@ webhooks:
       service:
         name: webhook-service
         namespace: '{{ .Release.Namespace }}'
-        path: /validate-app-fybrik-io-v1alpha1-fybrikmodule
+        path: /validate-app-fybrik-io-v1-fybrikmodule
     failurePolicy: Fail
-    name: vfybrikmodule.kb.io
+    name: vfybrikmodule-beta.kb.io
```
Comment (shlomitk1) [inline charts/fybrik/files/webhook-configs.yaml:40]: v1?

File: manager/controllers/app/fybrikapplication_controller_test.go [hunk 13]
```diff
@@ -15,7 +15,7 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 
-	apiv1alpha1 "fybrik.io/fybrik/manager/apis/app/v1alpha1"
+	app "fybrik.io/fybrik/manager/apis/app/v1"
```
Comment (shlomitk1) [inline manager/controllers/app/fybrikapplication_controller_test.go:18]: api, "app" is the current package.

File: samples/rest-server/Deployment.yaml [hunk 14]
```diff
@@ -0,0 +1,21 @@
+kind: Deployment
```
Comment (shlomitk1) [inline samples/rest-server/Deployment.yaml:1]: Why is this file committed?

File: pkg/optimizer/test.fzn [hunk 15]
```diff
@@ -0,0 +1,9 @@
+float: pi = 3.1415;
```
Comment (shlomitk1) [inline pkg/optimizer/test.fzn:1]: Why is this file committed?

File: manager/controllers/app/fybrikapplication_controller_test.go (hunk 13 above)
Comment (roytman) [reply manager/controllers/app/fybrikapplication_controller_test.go:18]: correct, fixed

File: charts/fybrik/files/webhook-configs.yaml (hunk 12 above)
Comment (roytman) [reply charts/fybrik/files/webhook-configs.yaml:40]: yes

File: charts/fybrik/files/webhook-configs.yaml (hunk 11 above)
Comment (roytman) [reply charts/fybrik/files/webhook-configs.yaml:19]: yes

File: manager/apis/app/v1/interface_types.go [hunk 16]
```diff
@@ -0,0 +1,19 @@
+// Copyright 2020 IBM Corp.
```
Comment (shlomitk1) [inline manager/apis/app/v1/interface_types.go:1]: I have deleted this file when cleaning apis.

File: manager/controllers/app/policy_compiler_interface.go [hunk 17]
```diff
@@ -59,7 +59,7 @@ func ValidatePolicyDecisionsResponse(response *policymanager.GetPolicyDecisionsR
 	}
 
 	return apierrors.NewInvalid(
-		schema.GroupKind{Group: "app.fybrik.io", Kind: "PolicyManager-GetPolicyDecisionsResponse"},
+		schema.GroupKind{Group: "fapp.fybrik.io", Kind: "PolicyManager-GetPolicyDecisionsResponse"},
```
Comment (shlomitk1) [inline manager/controllers/app/policy_compiler_interface.go:62]: Revert this change

File: manager/apis/app/v1/interface_types.go (hunk 16 above)
Comment (roytman) [reply manager/apis/app/v1/interface_types.go:1]: ok, probably during the rebase it slid into. Thanks

File: manager/controllers/app/policy_compiler_interface.go (hunk 17 above)
Comment (roytman) [reply manager/controllers/app/policy_compiler_interface.go:62]: thanks

**Comment by roytman** [comment]
> This should definitely be flagged as changing interfaces

Correct

File: charts/fybrik/files/webhook-configs.yaml [hunk 18]
```diff
@@ -14,14 +14,14 @@ webhooks:
       service:
         name: webhook-service
         namespace: '{{ .Release.Namespace }}'
-        path: /validate-app-fybrik-io-v1alpha1-fybrikapplication
+        path: /validate-app-fybrik-io-v1beta1-fybrikapplication
     failurePolicy: Fail
-    name: vfybrikapplication.kb.io
+    name: vfybrikapplication-v1.kb.io
```
Comment (shlomitk1) [inline charts/fybrik/files/webhook-configs.yaml:19]: v1beta1?

File: site/docs/samples/delete-sample.md [hunk 19]
```diff
@@ -93,7 +93,7 @@ Then, register the data asset itself in the data catalog `katalog` used for samp
 
 ```yaml
 cat << EOF | kubectl apply -f -
-apiVersion: katalog.fybrik.io/v1alpha1
+apiVersion: katalog.fybrik.io/v1
```
Comment (shlomitk1) [inline site/docs/samples/delete-sample.md:96]: katalog is still alpha

File: charts/fybrik/files/webhook-configs.yaml (hunk 18 above)
Comment (roytman) [reply charts/fybrik/files/webhook-configs.yaml]: Wow, thanks, fixed

File: site/docs/samples/delete-sample.md (hunk 19 above)
Comment (roytman) [reply site/docs/samples/delete-sample.md]: fixed

File: samples/rest-server/Deployment.yaml (hunk 14 above)
Comment (roytman) [reply samples/rest-server/Deployment.yaml]: removed

File: pkg/optimizer/test.fzn (hunk 15 above)
Comment (roytman) [reply pkg/optimizer/test.fzn]: removed

File: samples/multicluster-razee/README.md [hunk 20]
```diff
@@ -266,7 +266,7 @@ And then, register the data asset itself in the catalog on control plane. Replac
 kubectl config use-context kind-control
 
 cat << EOF | kubectl apply -f -
-apiVersion: katalog.fybrik.io/v1alpha1
+apiVersion: katalog.fybrik.io/v1beta1
```
Comment (shlomitk1) [inline samples/multicluster-razee/README.md:269]: katalog stays v1alpha1

Comment (roytman) [reply samples/multicluster-razee/README.md]: fixed

**Review by shlomitk1 — APPROVED** [review]
